# MEEN 689 HOMEWORK 6
Name: Chinonyelum Okereafor

UIN: 733001665

Datasets are included in the Excel file: HW6Data.xlsx

1. A study is conducted to determine if the wear rate is different for two brands of radial
automotive tires. One tire of each brand is mounted at random to the rear wheels of eight
cars. The tests continue until the tires wear out – and at that point the mileage of the car is
recorded. The following table shows this data. Consider the following for 95% confidence.
(Note: you may use Minitab, Excel, or R)

In [76]:
#Importing all libraries
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import scipy.stats as stats
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols
import statsmodels.api as sm
from statsmodels.formula.api import ols
from itertools import combinations
from statsmodels.stats.outliers_influence import variance_inflation_factor
import statsmodels.api as sm
from statsmodels.stats.anova import anova_lm
from statsmodels.stats.power import TTestIndPower
from scipy.stats import chi2

import warnings
warnings.filterwarnings("ignore")

In [77]:
#Reading the Excel file Q1 sheet into Python
df1 = pd.read_excel('/content/drive/MyDrive/1-School stuffs/3rd year/Fall 2024/MEEN 689/Homework/Homework 6/HW6Data.xlsx',
                    sheet_name = 'Q1')
print("Sheet Q1 Table:")
df1

Sheet Q1 Table:


Car  Brand “X”  Brand “Y”
0    1      36925      34318
1    2      45300      42280
2    3      36240      35500
3    4      32100      31950
4    5      37210      38015
5    6      48360      47800
6    7      38200      37810
7    8      33500      33215

a. Identify the type of t-test to be used and explain.

#Answers
Type of T-test: **Paired samples t-test using two-tailed** (also known as a dependent samples t-test or paired t-test)

**Explaination**

**Paired samples t-test:** Since each car tests both tire brands, each pair of mileage values (Brand "X" and Brand "Y") is dependent. The paired t-test is suitable here, as it evaluates whether the average difference in wear (mileage) between the brands is significantly different from zero.

**Two-tailed test:** From the question,we're checking if there’s a **difference** in wear rate between two tire brands, without specifying direction. Two-tailed test is appropriate because the difference could favor either brand. At 95% confidence, we would examine both distribution tails for statistical significance in either direction.

b. Calculate the degrees of freedom, t-value from the table, and t-value from the data.

In [78]:
#Data from the Datasheet Q1
brand_x = df1['Brand “X”']
brand_y = df1['Brand “Y”']

# Calculate the differences
differences = brand_x - brand_y

# Degrees of freedom (n - 1 for paired samples t-test)
n = len(differences)
degrees_of_freedom = n - 1

# Calculate t-value from the data
mean_diff = np.mean(differences)
std_diff = np.std(differences, ddof=1)
t_value_data = mean_diff / (std_diff / np.sqrt(n))

# Find the critical t-value from the table for 95% confidence level (two-tailed)
t_value_table = stats.t.ppf(1 - 0.05/2, degrees_of_freedom)


#Displaying results
print("================================================================")
print("Results for Two-tailed Paired samples t-test Test:")
print("----------------------------------------------------------------")
print("Degrees of Freedom:", degrees_of_freedom)
print("Mean Difference:", mean_diff)
print("Standard Deviation of Differences:", std_diff)
print("t-value from the data/sample:", t_value_data)
print("t-value from the table (Critical t-value):", t_value_table)
print("===============================================================")

Results for Two-tailed Paired samples t-test Test:
----------------------------------------------------------------
Degrees of Freedom: 7
Mean Difference: 868.375
Standard Deviation of Differences: 1290.0331045032472
t-value from the data/sample: 1.903932074206155
t-value from the table (Critical t-value): 2.3646242515927844


c. State the conclusion regarding the claim that the two brands of tires have the same wear
rate.

#Conclusion
Since the calculated t-value from the data (1.904) is less than the critical t-value from the table (2.365) at a 95% confidence level, **we fail to reject the null hypothesis (Not Rejecting H0).** This means there is no significance difference in the wear rates.

d. If these eight cars are identical (i.e., no car-to-car variation), would you use a different t-
test? If so, explain.

#Response
**Yes**, we would not a different t-test called **two-tailed independent two-sample t-test**.

**Explanation**

If the eight cars were identical with no variation between them, we would use the two-tailed independent two-sample t-test.

The independent two-sample t-test with equal population variances is appropriate for comparing the unpaired means of two independent groups (Brand "X" and Brand "Y" mileages) to determine if there is a significant difference in wear rates between the two brands.

e. Following d) to perform a new t-test and compare the result with c).

In [79]:
# Data for Brand X and Brand Y
brand_x = df1['Brand “X”']
brand_y = df1['Brand “Y”']

# Perform an independent two-sample t-test assuming equal variances
degrees_of_freedom_independent = len(brand_x) + len(brand_y) - 2
t_stat_independent, p_value_independent = stats.ttest_ind(brand_x, brand_y, equal_var=True)
t_stat_independent, p_value_independent
t_critical_independent = stats.t.ppf(1 - 0.05/2, degrees_of_freedom_independent)

print("================================================================")
print("Independent t-test Results:")
print("----------------------------------------------------------------")
print("t-statistic:", t_stat_independent)
print("p-value:", p_value_independent)
print("Critical t-value:", t_critical_independent)

if p_value_independent < 0.05 and t_stat_independent < t_critical_independent:
    print("Reject the null hypothesis: There is a significant difference in wear rates between the two brands.")
else:
    print("Conclusion --> Fail to reject the null hypothesis: There is no significant difference in wear rates between the two brands.")
print("===============================================================")

Independent t-test Results:
----------------------------------------------------------------
t-statistic: 0.3204417886417089
p-value: 0.7533662660432442
Critical t-value: 2.1447866879169273
Conclusion --> Fail to reject the null hypothesis: There is no significant difference in wear rates between the two brands.


#Comparing the Result to Question (1c):
The t-statistic is 0.3204 since the p-value (0.7534) is greater than 0.05. Also, since tsample (t-statistic) < ttable (t-critical), this implies that the **null hypothesis cannot be rejected** and this is consistent with the conclusion in question (1c) that there is NO significant difference in wear rates between the two brands.

2. A study is designed to test whether a surgical training course can change residents’
performance (in terms of time) on a certain procedure by at least 4 seconds. This study has
a control group (“not trained”) and a test group (“trained”) with an equal sample size.

In [80]:
#Reading the Excel file Q2 sheet into Python
df2 = pd.read_excel('/content/drive/MyDrive/1-School stuffs/3rd year/Fall 2024/MEEN 689/Homework/Homework 6/HW6Data.xlsx',
                    sheet_name = 'Q2')
print("Sheet Q2 Table:")
df2

Sheet Q2 Table:


Trained  Untrained
0       42         46
1       37         39
2       38         46
3       45         41
4       37         43
5       36         40
6       40         46
7       40         46
8       44         48

a. Assuming pooled standard deviation of 3 s, run the power analysis to determine the sample size needed for this study for an 80% power.

In [81]:
# Parameters for power analysis
Mean_difference = 4  # Given Mean difference (4 seconds)
pooled_std_dev = 3  # Given Pooled standard deviation (3 seconds)
effect_size = Mean_difference/pooled_std_dev
alpha = 0.05  # Significance level
power = 0.80  # Desired power level

# Perform power analysis to calculate sample size needed for each group
analysis = TTestIndPower()
sample_size = analysis.solve_power(effect_size=effect_size, power=power, alpha=alpha, alternative='two-sided')

print("================================================================")
print("Power Analysis Results:")
print("----------------------------------------------------------------")
print("Sample Size Needed for Each Group:", sample_size)
print("Approximated Sample Size:", np.ceil(sample_size))
print("===============================================================")

Power Analysis Results:
----------------------------------------------------------------
Sample Size Needed for Each Group: 9.889048179087231
Approximated Sample Size: 10.0


**Comments**

The power analysis indicates that approximately 10 participants per group are needed to achieve 80% power for detecting a mean difference of 4 seconds with a pooled standard deviation of 3 seconds, at a 5% significance level.

From the question, this is a two-tailed test. Therefore, using a two-sided test would be less efficient in this context requiring a larger sample size to maintain the same power and introducing the possibility of detecting a significant result in an irrelevant direction (i.e., worse performance by the trained group).

b. Datasheet Q2 includes the data collected for this study. Note the sample size may or
may not be the same as Part a. Run a proper t-test and conclude the result.

**Proper T-test**

In [82]:
# Calculating variances of the trained and untrained groups
variance_trained = np.var(df2['Trained'], ddof=1)  # Sample variance
variance_untrained = np.var(df2['Untrained'], ddof=1)  # Sample variance

print("================================================================")
print("Variance Results:")
print("----------------------------------------------------------------")
print(f"Variance of Trained Group: {variance_trained:.6f}")
print(f"Variance of Untrained Group: {variance_untrained:.6f}")
print("--->Therefore, the variances are equal")
print("===============================================================")

Variance Results:
----------------------------------------------------------------
Variance of Trained Group: 10.361111
Variance of Untrained Group: 10.361111
--->Therefore, the variances are equal


In [83]:
# Data for trained and untrained groups
trained = df2['Trained']
untrained = df2['Untrained']

# Calculate t-critical value for a two-tailed test with degrees of freedom based on sample size
n_trained = len(trained)
n_untrained = len(untrained)

#Mean difference
mean_diff = np.mean(untrained) - np.mean(trained)

degrees_of_freedom = n_trained + n_untrained - 2

# t-critical for 95% confidence level (two-tailed)
t_critical = stats.t.ppf(1 - 0.05/2, degrees_of_freedom)

# Perform an independent samples t-test
t_statistic, p_value = stats.ttest_ind(trained, untrained, equal_var=True)

print("================================================================")
print("Independent Samples t-test Results:")
print("----------------------------------------------------------------")
print("Mean Difference:", mean_diff)
print("t-statistic:", t_statistic)
print("p-value:", p_value)
print("Critical t-value:", t_critical)
print("===============================================================")

Independent Samples t-test Results:
----------------------------------------------------------------
Mean Difference: 4.0
t-statistic: -2.6361072933775653
p-value: 0.017971050809126213
Critical t-value: 2.1199052992210112


**Conclusion**

Since the absolute t-statistic (2.636) exceeds the t-critical value (2.120), **we reject the null hypothesis.** This confirms that the training course has a statistically significant effect on performance time.

c. What is the statistical power for the mean difference found in Part b. Note, the power
computed after the experiment is called post-hoc power.

In [84]:
pooled_std = np.sqrt((variance_trained + variance_untrained) / 2)  # since variances are equal

# Effect size based on observed mean difference
observed_effect_size = mean_diff / pooled_std

# Calculate post-hoc power
post_hoc_power = analysis.power(effect_size=observed_effect_size, nobs1=n_trained, alpha=0.05, alternative='two-sided')

print("================================================================")
print("Post-Hoc Power Results:")
print("---------------------------------------------------------------")
print("Post-Hoc Power:", post_hoc_power)
print("Post-Hoc Power (in %):",post_hoc_power*100,"%")
print("Approximated Post-Hoc Power (in %)",np.ceil(post_hoc_power*100),"%")
print("===============================================================")
print("Pooled Standard Deviation:", pooled_std)
print("Observed Effect size:", observed_effect_size)
print("===============================================================")

Post-Hoc Power Results:
---------------------------------------------------------------
Post-Hoc Power: 0.696934449434764
Post-Hoc Power (in %): 69.69344494347641 %
Approximated Post-Hoc Power (in %) 70.0 %
Pooled Standard Deviation: 3.2188679859713276
Observed Effect size: 1.2426728953883945


**Comments**

The post-hoc statistical power for the mean difference observed in part (b) , with a calculated pooled standard deviation of 3.22, the post-hoc power is about 69.7%.

This lower power suggests a higher likelihood of Type II error (failing to detect a true effect) than initially expected which indicates the study may **NOT** have had sufficient power to reliably detect the observed effect between the trained and untrained groups.

d. Is it necessary that post-hoc power (Part c) to be the same as priori power (Part a)? Do literature search and explain.

#Response to Question 1d
No, post-hoc power (calculated after the experiment based on observed effect size) is not necessarily the same as priori power (calculated before the experiment based on an expected effect size).



**Literature Search**

Research highlights that post-hoc power has limited practical value, as it relies heavily on the observed effect size, which can vary due to sampling. According to Hoenig and Heisey (2001), their study suggested that post-hoc power does not reliably assess a study's design or sample size adequacy. Instead, emphasis should be on a priori power to ensure a robust study design.

In summary, post-hoc and a priori power serve distinct purposes and need not align, as they are influenced by different factors in study planning.

.

**Explanation**

**Priori power** is calculated before a study begins to estimate the sample size needed to detect a desired effect, usually aiming for a power level like 80%. This ensures there is a high chance of detecting a true effect if one exists, guiding study planning.

**Post-hoc power** is calculated after data collection based on the actual effect size observed. It shows the study’s power to detect the effect that occurred not the initially expected effect.

Since the observed effect size often varies from the anticipated effect due to sampling variability, post-hoc power can differ from priori power thereby leading it to be higher or lower than planned.


**References**

Hoenig, J. M., & Heisey, D. M. (2001). The Abuse of Power: The Pervasive Fallacy of Power Calculations for Data Analysis. The American Statistician, 55(1), 19–24.

3. Datasheet (Q3) below data obtained by examining the success/failure of products being
produced on one of five different process “lines.” The question is whether the process line affects the
success/failure of the product?

In [85]:
#Reading the Excel file Q1 sheet into Python
df3 = pd.read_excel('/content/drive/MyDrive/1-School stuffs/3rd year/Fall 2024/MEEN 689/Homework/Homework 6/HW6Data.xlsx',
                    sheet_name = 'Q3')
print("Sheet Q3 Table:")
df3

Sheet Q3 Table:


Process Line  Success  Failure
0             1       31        7
1             2       57       33
2             3       87       60
3             4       52       44
4             5       10       19

a. Calculate expected frequencies and determine χ2obs. Show your calculations.

In [86]:
# Given data
data = {
    "Process Line": df3['Process Line'],
    "Success": df3['Success'],
    "Failure": df3['Failure']
}
df = pd.DataFrame(data)

# Total successes, failures, and grand total
total_success = df["Success"].sum()
total_failure = df["Failure"].sum()
grand_total = total_success + total_failure

# Calculating expected frequencies for each cell in a 5x2 table
df["Expected Success"] = ((df["Success"] + df["Failure"]) * total_success) / grand_total
df["Expected Failure"] = ((df["Success"] + df["Failure"]) * total_failure) / grand_total

# Calculating chi-square observed value
chi_square_obs = (((df["Success"] - df["Expected Success"]) ** 2) / df["Expected Success"]).sum() + \
                 (((df["Failure"] - df["Expected Failure"]) ** 2) / df["Expected Failure"]).sum()


print("=================================================================================")
print("Calculated Expected Frequencies:")
print("---------------------------------------------------------------------------------")
print(df[['Process Line', 'Success', 'Failure', 'Expected Success', 'Expected Failure']])
print("---------------------------------------------------------------------------------")
print("Total Success:", total_success)
print("Total Failure:", total_failure)
print("Grand Total:", grand_total)
print("=================================================================================")
print("Chi-square Observed Value:", chi_square_obs)
print("=================================================================================")

Calculated Expected Frequencies:
---------------------------------------------------------------------------------
   Process Line  Success  Failure  Expected Success  Expected Failure
0             1       31        7           22.5150           15.4850
1             2       57       33           53.3250           36.6750
2             3       87       60           87.0975           59.9025
3             4       52       44           56.8800           39.1200
4             5       10       19           17.1825           11.8175
---------------------------------------------------------------------------------
Total Success: 237
Total Failure: 163
Grand Total: 400
Chi-square Observed Value: 16.86401959592454


b. Determine the degrees of freedom and the X2crit

In [87]:
# Degrees of freedom for chi-square test in a contingency table
# (rows - 1) * (columns - 1)
rows = 5  # process lines
columns = 2  # success and failure
degrees_of_freedom = (rows - 1) * (columns - 1)

# Chi-square critical value for 95% confidence level
alpha = 0.05
chi_square_critical = stats.chi2.ppf(1 - alpha, degrees_of_freedom)

print("================================================================")
print("Degrees of Freedom:", degrees_of_freedom)
print("Chi-square Critical Value:", chi_square_critical)
print("===============================================================")

Degrees of Freedom: 4
Chi-square Critical Value: 9.487729036781154


c. Provide a statement that summarizes your findings.

#Summary of findings
Since the observed chi-square value (χ²ₒᵦₛ = 16.864) is greater than the critical chi-square value (χ²ₑᵢᵢₜ = 9.488) at a 95% confidence level, **the null hypothesis may be rejected**. This indicates that the process line does have a statistically significant effect on the success or failure of the product.

d. Use R or Minitab to run the same analysis and check the results.

In [88]:
# Running the chi-square test using Python's built-in function to verify results
# Contingency table for observed frequencies
observed = df[["Success", "Failure"]].values

# Chi-square test
chi2_stat, p_value, dof, expected = stats.chi2_contingency(observed)

# Calculate the chi-square critical value for the given alpha and degrees of freedom
chi2_critical = chi2.ppf(1 - alpha, dof)

expected_df = pd.DataFrame(expected, columns=["Expected Success", "Expected Failure"])
expected_df.insert(0, "Process Line", df["Process Line"])

print("================================================================")
print("Chi-square Test Results:")
print("----------------------------------------------------------------")
print("Chi-square Statistic:", chi2_stat)
print("Chi-square Critical Value:", chi2_critical)
print("p-value:", p_value)
print("Degrees of Freedom:", dof)
print("Expected Frequencies:")
print("---------------------------------------------------------------")
print(expected_df)
print("===============================================================")

Chi-square Test Results:
----------------------------------------------------------------
Chi-square Statistic: 16.86401959592454
Chi-square Critical Value: 9.487729036781154
p-value: 0.002054134688375616
Degrees of Freedom: 4
Expected Frequencies:
---------------------------------------------------------------
   Process Line  Expected Success  Expected Failure
0             1           22.5150           15.4850
1             2           53.3250           36.6750
2             3           87.0975           59.9025
3             4           56.8800           39.1200
4             5           17.1825           11.8175


**Comments**

Since the p-value (0.002) is less than 0.05, we conclude that the process line **significantly affects** product success or failure which is consistent with our previous findings.

.

**Checking the Results**

Since χ2obs > χ2critical, **we reject the null hypothesis** indicating a significant association between the process line and the success/failure rate of the products.
Also, the results are similar to the manually calculated results as shown in Questions 3(a) and 3(b).